### PR TITLE
Use recommended vscode configuration in gitpod environments

### PR DESCRIPTION
**Pull request by yardenshoham**
To make go tests run in gitpod, the vscode settings.json must be in the correct place in the filesystem

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 tasks:
   - name: Setup
     init: |
+      ln --symbolic contrib/ide/vscode .vscode
```
Comment (wxiaoguang) [inline .gitpod.yml]: A question, should it use `cp` instead? In case the user would modify their local `.vscode`.

**Review by wxiaoguang — APPROVED** [review]

**Comment by silverwind** [comment]
Why not move it into the correct place? Are the settings in there too opinionated? Note, I'm not a vscode user, so I can't really comment on them.

**Comment by yardenshoham** [comment]
> Why not move it into the correct place? Are the settings in there too opinionated? Note, I'm not a vscode user, so I can't really comment on them.

I'm for placing it at the root

**Comment by wxiaoguang** [comment]
> Why not move it into the correct place? Are the settings in there too opinionated? Note, I'm not a vscode user, so I can't really comment on them.

I am against for putting the `.vscode` in the root. These settings don't satisfy  everyone, some people may want to  change these settings. If the `.vscode` is put in the root, then how could they handle these changed files when committing?

**Comment by yardenshoham** [comment]
These settings are project-specific, any user settings are in another file. Having this at the root helps vscode users and hurts no one. I can't think of any project-specific setting vscode users have that shouldn't be committed.

**Comment by wxiaoguang** [comment]
> These settings are project-specific, any user settings are in another file. Having this at the root helps vscode users and hurts no one. I can't think of any project-specific setting vscode users have that shouldn't be committed.

But what if some people modified `launch.json` with other commands or changed the `go.buildTags` in `settings.json`?

**Comment by yardenshoham** [comment]
Then they probably did it for good reason and should commit the modification

**Comment by wxiaoguang** [comment]
> Then they probably did it for good reason and should commit the modification

I do not think so. When I am using VSCode, sometimes I add temporary options for my local debug purpose.

**Comment by yardenshoham** [comment]
If it's temporary, then you would simply won't commit it

**Comment by wxiaoguang** [comment]
Then I have to manually deselect them when committing 😂  again and again.

So, leave an empty `.vscode` and prepare it for everyone locally is the most clear solution IMO.

**Comment by yardenshoham** [comment]
> Then I have to manually deselect them when committing 😂 again and again.

How is that any different than not committing say, debug log statements?

**Comment by wxiaoguang** [comment]
* `.vscode` in root: ignore them from time to time if there are local changes.
* `vscode` in `contrib`: only copy the directory when first setup if you need (the new developer should have spent enough time on reading the development guideline, a copy won't take too much time), then change it freely as you want. No need to spend more time on it.

----

*  debug log statements: sometimes they can be avoided. If  they can not  be avoided, then the time can not be saved.

**Review by delvh — APPROVED** [review]

**Review by silverwind — APPROVED** [review]
It's an improvement in any case.

**Pull request merged**